hr_holidays_validity_date/models/hr_leave.py

```diff
@@ -0,0 +1,55 @@
+#     Copyright (c) 2015 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+
+
+class HrLeaveType(models.Model):
+    _inherit = "hr.leave.type"
+
+    restrict_dates = fields.Boolean(
+        string="Restrict",
+        help="Check this if you want to forbid requesting "
+             "leaves outside this range, otherwise it will just "
+             "display a warning to the user."
+    )
+
+
+class HolidaysRequest(models.Model):
+    _inherit = "hr.leave"
+
+    warning_validity = fields.Char(
+        compute='_compute_warning_range',
+        oldname='warning',
+    )
+    restrict_dates = fields.Boolean(string='Restrict dates',
+                                    related='holiday_status_id.restrict_dates')
+
+    @api.depends('holiday_status_id', 'date_from', 'date_to')
+    def _compute_warning_range(self):
+        for rec in self:
+            if rec.holiday_status_id.validity_start and (
```

Comment:
Create variable and check if 1 time better?
>  if vstart and vstop and dfrom and dto and ( dfrom.date() < vstart or dto.date() > vstop):

Reply:
I don't see where I can reuse the if, in every iteration a record may have a different leave type or different attributes in general

Reply:
okay, you are right.

Reply:
okay. you are right.